jquery.tmpl.js

```diff
@@ -7,7 +7,18 @@
  * Dual licensed under the MIT or GPL Version 2 licenses.
  * http://jquery.org/license
  */
-(function( jQuery, undefined ){
+(function (factory) {
+	if (typeof define === 'function' && define.amd) {
+		// AMD
+		define(['jquery'], factory);
+	} else if (typeof exports === 'object') {
+		// CommonJS
+		factory(require('jquery'));
+	} else {
+		// Browser globals
+		factory(jQuery);
+	}
+}(function( jQuery, undefined ){
```

Comment:
When loaded by AMD loaders, I'm not sure we can be certain that the second parameter is undefined, so we should probably write `}(function(jQuery) {`

So the whole thing could look like this:

```
(function( factory ) {
	if (typeof define === 'function' && define.amd) {
		// Loading from AMD script loader. Register as an anonymous module.
		define( ['jquery'], factory );
	} else {
		// Browser using plain <script> tag
		factory( jQuery );
	}
}(function( jQuery ){
...
```